src/jdkInstaller.ts

```diff
@@ -0,0 +1,97 @@
+import * as path from "path";
+import * as request from "request";
+import * as fs from "fs";
+import * as mkdirp from "mkdirp";
+import * as pcp from "promisify-child-process";
+import { OutputChannel } from "vscode";
+
+export class JdkInstaller {
+  out: OutputChannel;
+
+  homedir = require("os").homedir();
+  bin = path.join(this.homedir, "bin");
+
+  constructor(out: OutputChannel) {
+    this.out = out;
+  }
+
+  installJava(javaVersion: string, jabbaVersion: string) {
+    const jabbaUrl = `https://github.com/shyiko/jabba/releases/download/${jabbaVersion}/jabba-${jabbaVersion}-${this.jabbaUrlSuffix()}`;
+    const jabba = path.join(this.bin, this.jabbaName());
+    return new Promise<void>((res, rej) => {
+      mkdirp(this.bin, (err, made) => {
```

Comment:
minor, but we can probably use Node's `promisify` method here:


```
promisify(mkdirp)(this.bin)
```

I can work on this as well